test/beetle/configuration_test.rb

```diff
@@ -68,4 +68,51 @@ class ConfigurationTest < Minitest::Test
       assert_equal "10.0.0.1:3001", config.additional_subscription_servers
     end
   end
+
+  class ConnectionOptionsForServerTest < Minitest::Test
+
+    test "returns the options for the server provided" do
+      config = Configuration.new
+      config.servers = 'localhost:5672'
+      config.server_connection_options["localhost:5672"] = {host:  'localhost', port: 5672, user: "john", pass: "doe", vhost: "test", ssl: "0"}
```

Comment:
I might not be obvious why the value for `ssl` is `0` rather than true or false, but that is actually how the underlying amqp and bunny implementations need it.